src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
Your version is shorter, but not very readable. It is better to announce changes for conditions separately.
<pre>
const isMale = (person) => person.sex === 'm';
const isMother = (person) => people.some((m) => m.name === person.mother);

const children = onlyWithSon
  ? people.filter((person) => isMale(person) && isMother(person))
  : people.filter(isMother);
</pre>